testutil/base.go

```diff
@@ -40,20 +44,27 @@ func Attributes(t testing.TB) []sdk.Attribute {
 
 }
 
+// Resources produces an attribute list for populating a Group's
+// 'Resources' fields.
 func Resources(t testing.TB) []dtypes.Resource {
 	t.Helper()
 	count := rand.Intn(10) + 1
 
 	vals := make([]dtypes.Resource, 0, count)
 	for i := 0; i < count; i++ {
+		coin := sdk.NewCoin(CoinDenom, sdk.NewInt(int64(rand.Uint16())))
+		if coin.Amount.Int64() > int64(10000) {
+			limited := coin.Amount.Int64() % int64(1000)
+			coin = sdk.NewCoin(CoinDenom, sdk.NewInt(limited))
+		}
```

Comment:
Is this to address the fluctuating coverage?

I think that the action item on those is to make tests that target the sometimes-covered lines.

Reading through this, it's unclear what the point of capping it at 1,000 if it's over 10,000 is

Reply:
There's an error which pops up if the value is over 10k by some tests, I'm not exactly sure the reasoning.

So if it's over 10k, it gets remaindered by 1000 to make sure it's below 10k.

Reply:
why not just use `rand.Intn(10000)` initially?

Reply:
That seems like the obvious solution :)  Done.